app/controllers/application_controller.rb

```diff
@@ -1,10 +1,143 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+  get '/' do
+  	erb :index
+  end
+
+  get '/tweets' do
+  	if logged_in?
+  		@tweets = Tweet.all
+  		erb :'tweets/tweets'
+  	else
+  		redirect '/login'
+  	end
+  end
+
+  get '/tweets/new' do
+  	if logged_in?
+  		erb :'/tweets/create_tweet'
+  	else
+  		redirect '/login'
+  	end
+  end
+
+  post '/tweets' do
+  	if params[:content] != ""
+	  	@tweet = Tweet.new(params)
+	  	@tweet.user = current_user
+	  	@tweet.save
+	  	redirect '/tweets'
+	  end
+	  redirect '/tweets/new'
+  end
+
+  get '/signup' do
+  	if !logged_in?
+  		erb :'users/create_user'
+  	else
+  		redirect '/tweets'
+  	end
+  end
+
+  post '/signup' do
+	  	if params[:username] != "" && params[:email] != "" && params[:password] != ""
+	  		@user = User.create(params)
+	  		session[:id] = @user.id
+	  		redirect '/tweets'
+	  	end
+  	redirect '/signup'
+  end
+
+  get '/login' do
+  	if !logged_in?
+  		erb :'users/login'
+  	else
+  		redirect '/tweets'
+  	end
+  end
+
+  post '/login' do
+  	@user = User.find_by(username: params[:username])
+  	if @user
+  		session[:id] = @user.id
+  		redirect '/tweets'
+  	end
+  	redirect '/login'
+  end
+
+  get '/logout' do
+  	if logged_in?
+	    session.clear
+	    redirect "/login"
+	  else
+	  	redirect '/'
+	  end
+  end
+
+  get "/users/:slug" do
+  	@user = User.find_by_slug(params[:slug])
+  	erb :'users/tweets'
+  end
+
+  get '/tweets/:id' do
+  	if logged_in?
+	  	@tweet = Tweet.find_by_id(params[:id])
+	  	erb :'tweets/show_tweet'
+	  else
+	  	redirect '/login'
+	  end
+  end
+
+  get '/tweets/:id/edit' do
+  	if logged_in?
+  		@tweet = Tweet.find_by_id(params[:id])
+  		if @tweet.user_id == current_user.id
+  			erb :'tweets/edit_tweet'
+  		else
+  			redirect '/tweets'
+  		end
+  	else
+  		redirect '/login'
+  	end
+
+  end
+
+  patch '/tweets/:id' do
+  	@tweet = Tweet.find_by_id(params[:id])
+  	if params[:content] != "" && @tweet.content != params[:content]	  	
+	  	@tweet.content = params[:content]
+	  	@tweet.save
+	  	redirect "/tweets/#{@tweet.id}"
+	  end
+	  redirect "/tweets/#{@tweet.id}/edit"
+  end
+
+  delete '/tweets/:id/delete' do
+  	@tweet = Tweet.find_by_id(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.